### ISSUE TITLE
Create new incident update

### ISSUE THREAD

**Issue by base-zero** [CachetHQ/Cachet#2351]
When adding incident updates it would be nice to have the option to update the component status too, rather than having to post the incident update then go edit the incident to change the component status.

**Comment by jbrooksuk** [comment]
This isn't so straightforward. Would you preselect the components that would be affected? Would all components have their status changed?

**Comment by base-zero** [comment]
So based on the components which has been select in the incident, they would be pull into the update incident system so that we can change the status on them there while posting our update.

**Comment by jbrooksuk** [comment]
Ah, I see! Sorry :)

**Comment by gonzoinc** [comment]
I just finished implementing this on my instance. Looking through the code I see references in the UpdateIncidentCommandHandler.php to handle updating the component using UpdateComponetCommand, but it looks as though this does nothing, at least as far as I could see. So assuming unfinished code. I didn't want to mess with it so I just copied the component section from the create incident view into the update view and made simple DB update.

Starting with the view: resources/views/dashboards/incidents/update.blade.php after line 51
``` 
<input type="hidden" name="component_id" value="{{ $incident->component->id }}">
                <label>{{ trans('forms.incidents.component') }}</label>
                <div class="panel panel-default">
                    <div class="panel-body">
                        <div class="radio-items">
                            @foreach(trans('cachet.components.status') as $statusID => $status)
                            <div class="radio-inline">
                                <label>
                                    <input type="radio" name="component_status" value="{{ $statusID }}">
                                    {{ $status }}
                                </label>
                            </div>
                            @endforeach
                        </div>
                    </div>
                </div>
                <span class="help-block" style="margin-top:-20px">{{ trans('forms.optional') }}</span> 
```

This adds a hidden field to pass the componentId as I couldn't find it being passed in the incident object. The rest comes directly from the add.blade.php with the addition of the optional code block.

Update the handler app/Bus/Handlers/Commands/Incident/UpdateIncidentCommandHandler.php

Add the following into the top:

``` 
use Illuminate\Support\Facades\DB;
use Illuminate\Support\Facades\Log;
use Illuminate\Support\Facades\Input; 
```

This varies but right after the section commented "Update the component" add the following.

```  
$componentStatus = Input::get('component_status');
         
         if(isset($componentStatus) || !empty($componentStatus)) { 
             $results = DB::table('components')
                        ->where('id',$incident->component_id)
                        ->update(['status' => $componentStatus]);
                        
             if( $results != 1 ) {
                Log::error('Error: Component was not updated! ComponentId: ' . $incident->component_id . ' StatusID: ' . $componentStatus);
              } 
         }  
```

Explanation: K.I.S.S - Method "Keep It Simple Stupid" :p

View is self explanatory, Same as the other just adding in hidden component field needed to update it later on. 

Handler grabs the input value from the form, checks if it is set since it is optional, and updates the component status in the DB. Since the result should only ever return one if it doesn't we log it as an error and move on. This is basically a quick and easy method. Sure there are better ways using some of the custom stuff they have added in, but I don't like to overcomplicate things if it's not needed.

**Only thing it does not do that I may look into adding, is the update form I think should preset the value of the component status, but with the way the form uses radio buttons and the loop makes it a bit more tricky. Thinking something along the lines of adding a check into the radio button loop that if the button value = the component status then set it to checked.  I need to look into it more....

![screen shot 2017-06-16 at 11 08 07 am](https://user-images.githubusercontent.com/27728728/27234133-efa164e0-5289-11e7-8b98-53d9f6eb8814.png)

**Comment by jbrooksuk** [comment]
@gonzoinc when can you start contributing PR's?

**Comment by gonzoinc** [comment]
Making sure I'm legally allowed to contribute. Waiting on final company approval.... :(  Have to go through all the correct processes and such. I know I can publicly comment as they are just suggestions, but PR's are a different story... Don't want to jeopardize my job. Sorry if it's causing headaches.

**Comment by jbrooksuk** [comment]
No problem! I'd much rather everything be above board. Do you expect to get approval?

**Comment by codeagencybe** [comment]
Any update on this topic?

**Comment by jbrooksuk** [comment]
Replacing with https://github.com/CachetHQ/Cachet/issues/2508

**Comment by jbrooksuk** [comment]
(See point 3)

**Comment by jbrooksuk** [comment]
No worries, thanks for letting me know :)
